### PR TITLE
[17.01] qrencode: backport release 4.0 to stable tree

### PR DESCRIPTION
**Pull request by dibdot**
Signed-off-by: Dirk Brenken <dev@brenken.org>

Maintainer: @jp-bennett
Compile tested: -
Run tested: -

Description:
* backport current qrencode library to support the "-I" option in stable tree as well.

### CODE REVIEW

File: libs/qrencode/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (C) 2006-2012 OpenWrt.org
+# Copyright (C) 2006-2017 OpenWrt.org
```
Comment (karlp) [inline libs/qrencode/Makefile]: no. You're not openwrt.org and can't claim copyrigth on it's behalf, and the general form of it (adding a change 5-6 years after the fact) would be 2006-2012,2017 anyway.

**Comment by karlp** [comment]
It seems unnecessary to make major version bumps in a package in the stable branch, _particularly_ given how close we are to a new release.

**Comment by dibdot** [comment]
It's a 1:1 backport of the original PR, so I didn't change anything beyond that. 
We have two apps in the luci master tree (fwknopd & travelmate), that are using this missing qrencode option. People that are coming from 17.01 complain, that the qrcode generation not works as expected. Of course they could install the current snapshot release package manually ...

**Comment by karlp** [comment]
are you saying that fwknopd and travelmate _in 17.01_ were udpated to versions that required a newer qrencode?  or are you saying that fwknopd and travelmate in master use the available features of the newer packages in master?

The latter is "yep, that's how things work" and the former means that the updates in fwknopd and travelmate were completely broken and should be reverted.

**Comment by jp-bennett** [comment]
I don't think the fwknop package is using the -I option in 17.01.  If I did lose my mind and introduce that sort of breakage, we should fix it, but otherwise I'm inclined to wait for the 18.04 release.

**Comment by jp-bennett** [comment]
Also, @dibdot When qrencode pulled my patch upstream, the dev removed the -I short flag.  If and when he eventually does another release, we'll need to use "--inline" instead of "-I".

**Comment by dibdot** [comment]
Both packages are OK in 17.01 - no worries. I've received 9 complains in three days of travelmate users which are mixing up 17.01 base with currrent travelmate master ... but I'm fine to close this one and let's wait for 18.x. I'll add a short notice in lede-support forum.

Thanks for your feedback.

**Comment by karlp** [comment]
heh, if they're mixing up 17.01 and master with packages, they're in for more surprises than backporting some things are going to fix.

**Pull request closed without merging**